client/components/common/tce-core/ElementDiscussion.vue

```diff
@@ -0,0 +1,123 @@
+    attach offset-y left>
+    <template v-slot:activator="{ on: menu }">
+      <v-tooltip open-delay="800" left>
+        <template v-slot:activator="{ on: tooltip }">
+          <v-btn v-on="{ ...menu, ...tooltip }" icon>
+            <v-badge
+              v-if="unseenCommentCount"
+              :content="unseenCommentCount"
+              color="secondary" />
+            <v-icon v-else color="primary" class="pr-1">mdi-forum-outline</v-icon>
+          </v-btn>
+        </template>
+        <span>Discussion</span>
+      </v-tooltip>
+    </template>
+    <discussion
+      @save="save"
+      @update="save"
+      @remove="editorBus.emit(events.REMOVE, $event)"
+      v-bind="{ comments, user }"
+      show-heading
+      class="pa-4" />
+  </v-menu>
+</template>
+
+<script>
+import Discussion from 'tce-core/Discussion';
+import discussionEvent from './Events/discussionEvent';
+import { mapChannels } from '@/plugins/radio';
+
+export default {
+  name: 'element-discussion',
+  props: {
+    id: { type: Number, required: true },
+    uid: { type: String, required: true },
+    comments: { type: Array, required: true },
+    lastSeen: { type: Number, required: true },
+    lastCommentAt: { type: Number, required: true },
+    user: { type: Object, required: true }
+  },
+  data: () => ({
+    isVisible: false,
+    unseenCommentCount: 0
+  }),
+  computed: {
+    ...mapChannels({ editorBus: 'editor' }),
+    events: () => discussionEvent,
+    unseenComments() {
+      const { comments, user, lastSeen } = this;
+      return comments.filter(it =>
+        it.authorId !== user.id &&
+        new Date(it.createdAt).getTime() > lastSeen
+      );
+    }
+  },
+  methods: {
+    save(data) {
+      const { editorBus, user: author, id: elementId, events } = this;
+      return editorBus.emit(events.SAVE, {
+        ...data,
+        author,
+        contentElementId: elementId
+      });
+    },
+    setLastSeen(timeout) {
+      const { uid: elementUid, lastCommentAt, events } = this;
+      const options = { timeout, elementUid, lastCommentAt };
+      this.editorBus.emit(events.SET_LAST_SEEN, options);
+    }
+  },
+  watch: {
+    isVisible(val) {
+      if (!val || !this.lastCommentAt) return;
+      this.setLastSeen(1000);
+    },
+    comments(val, oldVal) {
+      if (!this.isVisible || val === oldVal) return;
+      this.setLastSeen(2000);
+    },
+    unseenComments(comments) {
+      if (this.isVisible && comments.length) return;
+      setTimeout(() => (this.unseenCommentCount = comments.length), 200);
+    }
+  },
+  components: { Discussion }
+};
+</script>
+
+<style lang="scss" scoped>
+$background-color: #fff;
+
+.v-menu__content {
+  background: $background-color;
+
+  ::v-deep .embedded-discussion {
+    text-align: left;
+  }
+}
+
+.v-btn {
+  position: absolute;
+  top: 2rem;
+  right: -1.125rem;
+  z-index: 2;
+  background: $background-color;
+
+  .v-btn__content {
+    position: relative;
+  }
+
+  ::v-deep .v-badge {
```

Comment:
Not sure if this is the intention of the component when used within btn. Maybe we should use an avatar within btn with a number to avoid this.